src/components/NotificationStack/index.js

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { Notification as NotificationItem } from '@santiment-network/ui'
+import { CSSTransition, TransitionGroup } from 'react-transition-group'
+import styles from './NotificationStack.module.scss'
+
+const notifyDuration = +styles.notifyDuration
+
+class NotificationStack extends Component {
+  timerHandles = {}
+
+  state = {
+    notifications: []
+  }
+
+  static getDerivedStateFromProps ({ notification }, prevState) {
+    return {
+      notifications: notification
+        ? [...prevState.notifications, notification]
+        : []
+    }
+  }
+
+  componentWillUnmount () {
+    Object.keys(this.timerHandles).forEach(timerName => {
+      if (this.timerHandles[timerName]) {
+        clearTimeout(this.timerHandles[timerName])
+      }
+    })
+  }
+
+  componentDidUpdate (prevProps, prevState) {
+    if (this.state.notifications.length > prevState.notifications.length) {
+      const { id, dismissAfter } = this.state.notifications[
+        this.state.notifications.length - 1
+      ]
+
+      this.timerHandles[id] = setTimeout(() => {
```

Comment:
Much easier to save last notification into the static variable than doing line `33`-`36`